src/Transports.AspNetCore/GraphQLHttpMiddlewareOptions.cs

```diff
@@ -0,0 +1,107 @@
+#nullable enable
+
+using System.Security.Claims;
+using System.Security.Principal;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+
+namespace GraphQL.Server.Transports.AspNetCore;
+
+/// <summary>
+/// Configuration options for <see cref="GraphQLHttpMiddleware"/>.
+/// </summary>
+public class GraphQLHttpMiddlewareOptions
+{
+    /// <summary>
+    /// Enables handling of GET requests.
+    /// </summary>
+    public bool HandleGet { get; set; } = true;
+
+    /// <summary>
+    /// Enables handling of POST requests, including form submissions, JSON-formatted requests and raw query requests.
+    /// </summary>
+    public bool HandlePost { get; set; } = true;
+
+    /********** WebSockets support ********
+    
+    /// <summary>
+    /// Enables handling of WebSockets requests.
+    /// <br/><br/>
+    /// Requires calling <see cref="WebSocketMiddlewareExtensions.UseWebSockets(IApplicationBuilder)"/>
+    /// to initialize the WebSocket pipeline within the ASP.NET Core framework.
+    /// </summary>
+    public bool HandleWebSockets { get; set; } = true;
+
+    ******************/
+
+    /// <summary>
+    /// Enables handling of batched GraphQL requests for POST requests when formatted as JSON.
+    /// </summary>
+    public bool EnableBatchedRequests { get; set; } = true;
+
+    /// <summary>
+    /// Enables parallel execution of batched GraphQL requests.
+    /// </summary>
+    public bool BatchedRequestsExecuteInParallel { get; set; } = true;
+
+    /// <summary>
+    /// When enabled, GraphQL requests with validation errors
+    /// have the HTTP status code set to 400 Bad Request.
+    /// GraphQL requests with execution errors are unaffected.
+    /// <br/><br/>
+    /// Does not apply to batched or WebSocket requests.
+    /// </summary>
+    public bool ValidationErrorsReturnBadRequest { get; set; } = true;
+
+    /// <summary>
+    /// Enables parsing the query string on POST requests.
+    /// If enabled, the query string properties override those in the body of the request.
+    /// </summary>
+    public bool ReadQueryStringOnPost { get; set; } = true;
+
+    /// <summary>
+    /// Enables reading variables from the query string.
+    /// Variables are interpreted as JSON and deserialized before being
+    /// provided to the <see cref="IDocumentExecuter"/>.
+    /// </summary>
+    public bool ReadVariablesFromQueryString { get; set; } = true;
```

Comment:
I would add some comment why disabling it may be useful. Security reasons?

Reply:
The same for `ReadExtensionsFromQueryString`.

Reply:
Go ahead and post a suggestion if you like.  Perhaps something about how variables on the query string may be logged as part of the url, or something.  I wouldn't bother; if someone wants to change it, they will know why.

Reply:
>  I wouldn't bother

OK. I don't know particular reasons.